src/exceptions/api.error.js

```diff
@@ -0,0 +1,32 @@
+export class ApiError extends Error {
+  constructor({ message, status, errors }) {
+    super(message);
+
+    this.errors = errors;
+    this.status = status;
+  }
+
+  static badRequest(message, errors) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    });
+  }
+
+  static unathorized(errors) {
```

Comment:
There is a typo in the method name 'unathorized'. It should be 'unauthorized' to match the correct spelling.